hangman/game.py

```diff
@@ -1,23 +1,87 @@
 from .exceptions import *
+import random
 
 # Complete with your own, just for fun :)
-LIST_OF_WORDS = []
+LIST_OF_WORDS = ['cookies', 'cream', 'love', 'vinegar']
 
 
 def _get_random_word(list_of_words):
-    pass
+    if list_of_words == []:
+        raise InvalidListOfWordsException
+    else:
+        return random.choice(list_of_words)
 
 
 def _mask_word(word):
-    pass
+    if word == "":
+        raise InvalidWordException
+    else:
+        return len(word)*"*"
 
 
+def find_substring(string, substring):
+    """ 
+    Returns list of indices where substring begins in string
+    >>> find_substring('r', "rmotr")
+    [0, 4]
+    """
+    indices = []
+    index = -1  # Begin at -1 so index + 1 is 0
+    while True: 
+        # Find next index of substring, by starting search from index + 1
+        index = string.find(substring, index + 1)
+        if index == -1:  
+            break  # All occurrences have been found
+        indices.append(index)
+    return indices
+
 def _uncover_word(answer_word, masked_word, character):
-    pass
+    if character == "" or len(character) > 1 :
+        raise InvalidGuessedLetterException(Exception)
+    if len(masked_word) != len(answer_word):
+        raise InvalidWordException
+    if answer_word =="" or masked_word =="" :
+        raise InvalidWordException
+    #convert answer_word and character to lower case    
+    lower_character = str.lower(character)
+    lower_answer = str.lower(answer_word)
+    if lower_character not in lower_answer:
```

Comment:
we can remove line 48 and 49 since there's no change to masked_word ... notice line 55 will return masked_word anyway ...